Makefile

```diff
@@ -12,7 +12,11 @@ CMD_GITHUB ?= gh
 # environment:
 ARCH ?= $(shell uname -m)
 KERN_RELEASE ?= $(shell uname -r)
-KERN_SRC ?= /lib/modules/$(KERN_RELEASE)/build
+KERN_BLD_PATH ?= $(if $(KERN_SRC),$(KERN_SRC),/lib/modules/$(KERN_RELEASE)/build)
```

Comment:
KERN_SRC is used to provide the value for KERN_BLD_PATH? what's the reasoning behind this?

Reply:
There are two "flavors" of kernel headers, according to the distro.
One flavor is using one directory for all kernel headers, including generated. In this case KERN_SRC should be chosen.
The other flavor is when the headers are split into two directories - source and build.
If the user explicitly chose KERN_SRC, it should be used both for KERN_BLD_PATH and KERN_SRC_PATH, and override the automatic search for build and source paths.

Reply:
My misunderstanding was due to the names - KERN_SRC is not really the SOURCE rather the BUILD dir. This "fault" was also before but now it's standing out because we added separate vars for SOURCE and BUILD. 
Maybe instead of asking the user for the location of /build and /source we ask them for the location of /lib/modules/$version and we find /build or /source there?
From `linux/Makefile`:

```
MODLIB = $(INSTALL_MOD_PATH)/lib/modules/$(KERNELRELEASE)
...
ln -s $(abspath $(srctree)) $(MODLIB)/source
...
ln -s $(CURDIR) $(MODLIB)/build
```

Reply:
This doesn't take into consideration the case that the user doesn't have "source" and "build" sub directories.
For example, if the user downloaded the sources, and gives it as a parameter to KERN_SRC.

I think the name KERN_SRC is ok, but we can change it to KERN_HDRS to be more clear.
Then we will have:
KERN_HDRS - that points to a directory with all headers included.
KERN_SRC_PATH, KERN_BLD_PATH - that points to the source and build directories repsectively

Reply:
ok I understand. so isn't the KERN_SRC redundant? why don't we ask the user for the /source and the /build paths, and if the user doesn't give them we default, and if the user give just one we default the other one.

Reply:
Changed to KERN_HEADERS as discussed